be/src/storage/persistent_index.cpp

```diff
@@ -2840,6 +2871,29 @@ Status PersistentIndex::try_replace(size_t n, const Slice* keys, const IndexValu
     return Status::OK();
 }
 
+Status PersistentIndex::flush_advance() {
```

Comment:
If flush advance, how much degradation of apply phase?

Reply:
> If flush advance, how much degradation of apply phase?

It is related to the data size. In my test case(1 Billion rows with 20 buckets), the apply cost time increased from 736379(ms) to 924492(ms). And the degree of performance degradation is positively correlated with the amount of data imported.

The merge compaction could be changed to asyn, and it maybe finished in the future pr which can resolve the degradation problem.

Reply:
> > If flush advance, how much degradation of apply phase?
> 
> It is related to the data size. In my test case(1 Billion rows with 20 buckets), the apply cost time increased from 536379(ms) to 754727(ms). And the degree of performance degradation is positively correlated with the amount of data imported.
> 
> The merge compaction could be changed to asyn, and it maybe finished in the future pr which can resolve the degradation problem.

I did the following two optimizations:
1. Do `_merge_compaction` in advance to reduce the tmp l1 file to reduce the data read time in apply
2. Do not sync tmp l1 file during apply and only sync the final l1 file.
The apply cost time is reduced to 585722(ms) which is 754727(ms) without the optimizations.